app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,72 @@
+class TweetsController < ApplicationController
+
+  get '/tweets' do
+    if logged_in?
+      @user = current_user
+      erb :'tweets/tweets'
+    else
+      redirect to '/login'
+    end
+  end 
+
+  # Create --------------------------------------------------------------------
+
+  get '/tweets/new' do
+    logged_in? ? (erb :'tweets/create_tweet') : (redirect to '/login')
+  end 
+
+  post '/tweets' do
+    if params[:content].empty?
+      flash[:message] ="Provide some content to post a tweet!"
+      redirect to '/tweets/new'
+    elsif logged_in?
+      @user = current_user
+      @tweet =Tweet.create(params)
+      @user.tweets << @tweet
+      redirect to "tweets/#{@tweet.id}"
+    else
+      redirect to '/login'
+    end 
+  end
+
+  # Read ----------------------------------------------------------------------
+
+  get '/tweets/:id' do
+    redirect to '/login' if !logged_in?
+    @tweet = Tweet.find_by_id(params[:id])
+    erb :'tweets/show_tweet'
+  end 
+
+  # Update --------------------------------------------------------------------
+
+  get '/tweets/:id/edit' do 
+    redirect to '/login' if !logged_in?
+    @tweet = Tweet.find(params[:id])
+    if logged_in? && @tweet.user != current_user
+      flash[:notice] = "You cannot change another user's content, silly"
+      redirect to "/tweets/#{@tweet.id}"
+    end
+    erb :'tweets/edit_tweet'
+  end 
+
+  patch '/tweets/:id' do
+    @tweet = Tweet.find(params[:id])
+    redirect to "/tweets/#{@tweet.id}/edit" if params[:content].empty?
+    @tweet.update(content: params[:content])
+    redirect to "/tweets/#{@tweet.id}"
+  end 
+
+  # Delete --------------------------------------------------------------------
+
+  post '/tweets/:id/delete' do
+    @tweet = Tweet.find(params[:id])
+    if logged_in? && @tweet.user == current_user
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.